cli/chaosli/cmd/context.go

```diff
@@ -0,0 +1,388 @@
+	s.Start()
+
+	size, err := getTargetSize(disruption)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// Remove header NAME from consideration
+	size--
+	// If the size is 0, first check if changing the level will do anything, otherwise
+	// mention to the user that the labels they are using won't target anything
+
+	if size <= 0 {
+		errorString := fmt.Sprintf("\nThe label selectors chosen (%s) result in 0 targets, meaning this disruption would do nothing given the namespace/cluster/label combination.", labels)
+		newLevel := ""
+
+		if level == types.DisruptionLevelPod {
+			disruption.Spec.Level = types.DisruptionLevelNode
+		} else {
+			disruption.Spec.Level = types.DisruptionLevelPod
+		}
+
+		size, err = getTargetSize(disruption)
+
+		if err != nil {
+			return nil, err
+		}
+
+		if size > 0 {
+			errorString = fmt.Sprintf("\nWe noticed that your target size is 0 for level %s given your label selectors. We checked to see if the %s level would give you results and we found %d %ss. Is this the level you wanted to use?", level, newLevel, size, newLevel)
+		}
+
+		return nil, fmt.Errorf(errorString)
+	}
+
+	cmd := fmt.Sprintf("kubectl get %s -n %s -l %s | awk '{print $%d}'", level, podNamespaces, labels, rowNames)
+	targets, err := exec.Command("bash", "-c", cmd).Output()
+
+	if err != nil {
+		return nil, fmt.Errorf("could not grab list of targets names correlating to target selector: %v", err)
+	}
+
+	s.Stop()
+
+	targetsShow := []string{}
+	targetsAll := []string{}
+	targetsSplit := strings.Split(string(targets), "\n")
+
+	for i := 0; i < len(targetsSplit); i++ {
+		if len(targetsShow) < MAXTARGETSHOW {
+			targetsShow = append(targetsShow, targetsSplit[i])
+		}
+
+		if targetsSplit[i] == "NAME" || targetsSplit[i] == "" {
+			continue
+		}
+
+		targetsAll = append(targetsAll, targetsSplit[i])
+	}
+
+	fmt.Printf("\n🎯 There are %d %ss that will be targeted\n", size, level)
+
+	if size >= MAXTARGETSHOW {
+		fmt.Println("📝 Here are a small set of those targets:")
+	}
+
+	for _, target := range targetsShow {
+		fmt.Println(target)
+	}
+
+	if size >= MAXTARGETSHOW {
+		fmt.Println("...")
+	}
+
+	PrintSeparator()
+
+	return targetsAll, nil
+}
+
+func getTargetSize(disruption v1beta1.Disruption) (int, error) {
+	level := disruption.Spec.Level
+	podNamespace := disruption.Namespace
+	labels := disruption.Labels
+	cmd := fmt.Sprintf("kubectl get %s -n %s -l %s | wc -l", level, podNamespace, labels)
+	sizeString, err := exec.Command("bash", "-c", cmd).Output()
+
+	if err != nil {
+		return -1, fmt.Errorf("could not count the number of targets correlating to target selector: %v", err)
+	}
+
+	size, err := strconv.Atoi(strings.Trim(string(sizeString), "\n"))
+
+	if err != nil {
+		return -1, fmt.Errorf("could not convert string to integer in context target size: %v", err)
+	}
+
+	return size, nil
+}
+
+func grabDataForTargets(targets []string, disruption v1beta1.Disruption) ([]v1.Pod, []v1.Node, error) {
```

Comment:
@Devatoria whats the idiomatic way (if you know it) in Go to return A XOR B in this fashion?

Reply:
I think the current return value works fine then, though the function could benefit from a comment explaining that only one slice will be populated